### PR TITLE
Machineclass

### PR DESCRIPTION
**Pull request by enxebre**
Add support getting machineClasses from machine.Spec.ProviderConfig.ValueFrom
Needs https://github.com/openshift/cluster-api-provider-aws/pull/115

### CODE REVIEW

File: pkg/cloud/aws/actuators/machine/utils.go [hunk 1]
```diff
@@ -138,9 +140,28 @@ func terminateInstances(client awsclient.Client, instances []*ec2.Instance) erro
...
+		if machine.Spec.ProviderConfig.ValueFrom != nil && machine.Spec.ProviderConfig.ValueFrom.MachineClass != nil {
+			key := types.NamespacedName{
+				Namespace: machine.Spec.ProviderConfig.ValueFrom.MachineClass.Namespace,
+				Name:      machine.Spec.ProviderConfig.ValueFrom.MachineClass.Name,
+			}
+			if err := client.Get(context.Background(), key, machineClass); err != nil {
+				return nil, err
+			}
+			providerConfig = machineClass.ProviderConfig
+		}
```
Comment (ingvagabund) [inline pkg/cloud/aws/actuators/machine/utils.go]: Missing

```go
} else {
	return fmt.Errorf("unable to find machine provider config: neither Spec.ProviderConfig.Value nor Spec.ProviderConfig.ValueFrom set"), nil
}
```

File: pkg/cloud/aws/actuators/machine/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		},
+	}
+
+	client := fake.NewFakeClient(machineClass)
+	for _, tc := range testCases {
+		decodedProviderConfig, err := ProviderConfigFromMachine(client, tc.machine)
+		if err != nil {
+			t.Error(err)
+		}
+		if decodedProviderConfig.InstanceType != "testInstance" {
```
Comment (ingvagabund) [inline pkg/cloud/aws/actuators/machine/utils_test.go]: `reflect.DeepEqual` will do better so you don't have to test for concrete fields.

File: pkg/cloud/aws/actuators/machine/utils.go (hunk 1 above)
Comment (frobware) [reply pkg/cloud/aws/actuators/machine/utils.go]: Presumably the return arguments are they other way around here.

Comment (ingvagabund) [reply pkg/cloud/aws/actuators/machine/utils.go]: Yeah, you are right. Though, @enxebre implemented that case differently.

**Pull request merged**